src/lib/Commands/TestCommand/index.js

```diff
@@ -268,22 +282,35 @@ class TestCommand extends AbstractCommand {
     });
   }
 
-  /**
-   * Prepare source code
-   * @param testFile
-   * @return {{agent: string, device: string}}
-   * @private
-   */
+  // Prepare source code
+  // @param testFile
+  // @return {{agent: string, device: string}}
+  // @private
+
   _getSessionCode(testFile) {
     let agentCode, deviceCode;
 
-    /* [info] */
+    // [info]
     this._info(c.blue('Using ') + testFile.type + c.blue(' test file ') + testFile.name);
 
     // read/process test code
-    let testCode = fs.readFileSync(testFile.path, 'utf-8').trim();
-    this._codeProcessor.variables.__FILE__ = path.basename(testFile.path);
-    testCode = this._codeProcessor.process(testCode);
+    let testCode = fs.readFileSync(testFile.path, 'utf-8').trim()
+        .replace(LINE_REGEXP, "@{__LINE__}").replace(FILE_REGEXP, "@{__FILE__}");
+    var environmentVars = "";
+    for (var prop in process.env) {
+      if (prop !== BUILD_API_KEY_ENV_VAR) { //deny to access for BUILD_API_KEY_ENV_VAR
+        // Replace #{env:...} with @{...} if it is needed
+        let propertyRegExp = new RegExp('\\#\\{env:\\s*' + prop + '\\s*\\}', 'g');
+        if (testCode.match(propertyRegExp)) {
+          testCode = testCode.replace(propertyRegExp, '@{' + prop + '}');
+          environmentVars = environmentVars + "@set "+ prop + " \"" + process.env[prop] + "\"\n";
```

Comment:
and to be honest I believe we need to make Builder retrieving variables (if not set explicitly set in the code) from the environment, and avoid this dancing around the @set prop here. and it should be builder that swear on that

this needs to be discussed and thought over a little bit more and is probably worth filing an issue against the Builder.

BTW why don't you just replace the #{env:VARIABLE} with just it's "VALUE" instead of jumping up with @set/@{} expressions?